internal/resources/metal/vlans/resource.go

```diff
@@ -0,0 +1,186 @@
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+	return
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	vlan, _, err := client.ProjectVirtualNetworks.Get(
+		data.ID.ValueString(),
+		&packngo.GetOptions{Includes: []string{"assigned_to"}},
+	)
+	if err != nil {
+		if equinix_errors.IsNotFound(err) {
+			response.Diagnostics.AddWarning(
+				"Equinix Metal Vlan not found during refresh",
+				fmt.Sprintf("[WARN] Vlan (%s) not found, removing from state", data.ID.ValueString()),
+			)
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError("Error fetching Vlan using vlanId",
+			equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	response.Diagnostics.Append(data.parse(vlan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (r *Resource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	vlan, resp, err := client.ProjectVirtualNetworks.Get(
+		data.ID.ValueString(),
+		&packngo.GetOptions{Includes: []string{"instances", "meta_gateway"}},
+	)
+	if err != nil {
+		if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusNotFound {
+			response.Diagnostics.AddWarning(
+				"Equinix Metal Vlan not found during delete",
+				err.Error(),
+			)
+			return
+		}
+		response.Diagnostics.AddError("Error fetching Vlan using vlanId",
+			equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	// all device ports must be unassigned before delete
+	for _, instance := range vlan.Instances {
+		for _, port := range instance.NetworkPorts {
+			for _, v := range port.AttachedVirtualNetworks {
+				if v.ID == vlan.ID {
+					_, resp, err = client.Ports.Unassign(port.ID, vlan.ID)
+					if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusNotFound {
+						response.Diagnostics.AddError("Error unassign port with Vlan",
+							equinix_errors.FriendlyError(err).Error())
+						return
+					}
+				}
+			}
+		}
+	}
+
+	resp, err = client.ProjectVirtualNetworks.Delete(vlan.ID)
+	if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusNotFound {
```

Comment:
There should be an `err != nil` check in here

Reply:
On second glance it looks like this used to use `equinix_errors.IgnoreResponseErrors` and should continue to do that.

Reply:
Yes fixed.